proto/data_objects.proto

```diff
@@ -0,0 +1,268 @@
+  string id = 1;
+  string cloud = 2;
+}
+
+message DataObjectNativeURLResponse {
+  string url = 1;
+  string key = 2;
+  string secret_key = 3;
+}
+
+message AddDataBundleObjectResponse {
+  string id = 1;
+}
+
+message GetDataBundleObjectRequest {
+  string id = 1;
+}
+
+message GetDataBundleObjectChecksumRequest {
+  string checksum = 1;
+}
+
+message DeleteDataBundleObjectRequest {
+  string id = 1;
+}
+
+message DeleteDataObjectRequest {
+  string id = 1;
+}
+
+message DataBundleStatusResponse {
+  //REQUIRED
+  string id = 1;
+
+  //REQUIRED
+  repeated DataBundleObject data_bundle_versions = 2;
+}
+
+message DataBundleChecksumStatusResponse {
+  //REQUIRED
+  string checksum = 1;
+
+  //REQUIRED
+  repeated DataBundleObject data_bundle_versions = 2;
+}
+
+message ListDataBundleObjectsRequest {
+  //OPTIONAL
+  string id = 1;
+
+  //OPTIONAL
+  string dataset_id = 3;
+
+  //OPTIONAL
+  string submitted_id = 4;
+
+  //OPTIONAL
+  // Specifies the maximum number of results to return in a single page.
+  // If unspecified, a system default will be used.
+  int32 page_size = 5;
+
+  //OPTIONAL
+  // The continuation token, which is used to page through large result sets.
+  // To get the next page of results, set this parameter to the value of
+  // `next_page_token` from the previous response.
+  string page_token = 6;
+
+}
+
+message ListDataBundleObjectsResponse {
+  // The list of datasets.
+  repeated DataBundleObject data_bundle_objects = 1;
+
+  // The continuation token, which is used to page through large result sets.
+  // Provide this value in a subsequent request to return the next page of
+  // results. This field will be empty if there aren't any additional results.
+  string next_page_token = 2;
+}
+
+message FileStatusResponse {
+  //REQUIRED
+  string id = 1;
+
+  //REQUIRED
+  repeated DataObject file_versions = 2;
+
+}
+
+message GetDataObjectChecksumResponse {
+  //REQUIRED
+  string checksum = 1;
+
+  //REQUIRED
+  repeated DataObject file_versions = 2;
+
+}
+
+///-------------------------------------------------------------------
+// Resource: a file, API or other resource
+///-------------------------------------------------------------------
+message DataObject {
+  //REQUIRED
+  string id = 1;
+
+  //REQUIRED
+  string file_name = 2;
+
+  //REQUIRED
+  int64 file_size = 3;
+
+  // REQUIRED
+  // Timestamp of object creation in RFC3339.
+  google.protobuf.Timestamp created = 4;
+
+  // REQUIRED
+  // Timestamp of update in RFC3339, identical to create timestamp in systems
+  // that do not support updates.
+  google.protobuf.Timestamp updated = 5;
+
+  // REQUIRED
+  // A string representing a version, some systems may use checksum, a RFC3339
+  // timestamp, or incrementing version number. For systems that do not support
+  // versioning please use your update timestamp as your version.
+  string version = 6;
+
+  // OPTIONAL
+  string mime_type = 7;
+
+  // REQUIRED, the file SHA256 checksum
+  string checksum =  8;
```

Comment:
indexd supports multiple checksum types. need to resolve
https://github.com/ohsu-comp-bio/indexd/blob/master/indexd/index/schema.py#L35-L76

Reply:
Yeah, I think we need to use a data structure that combines an enum (checksum used) with the actual checksum.

Reply:
https://github.com/ga4gh/data-object-schemas/issues/12